packages/application-generic/src/services/readiness.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { BullMqService } from './bull-mq.service';
+
+import {
+  QueueServiceHealthIndicator,
+  TriggerQueueServiceHealthIndicator,
+  WsQueueServiceHealthIndicator,
+} from '../health';
+
+export interface INovuWorker {
+  name: string;
+  bullMqService: BullMqService;
+  gracefulShutdown: () => Promise<void>;
+  pauseWorker: () => Promise<void>;
+  resumeWorker: () => Promise<void>;
+}
+
+const LOG_CONTEXT = 'ReadinessService';
+
+@Injectable()
+export class ReadinessService {
+  constructor(
+    private queueServiceHealthIndicator: QueueServiceHealthIndicator,
+    private triggerQueueServiceHealthIndicator: TriggerQueueServiceHealthIndicator,
+    private wsQueueServiceHealthIndicator: WsQueueServiceHealthIndicator
+  ) {}
+
+  async areQueuesEnabled(): Promise<boolean> {
+    Logger.log('Enabling queues as workers are meant to be ready', LOG_CONTEXT);
+
+    try {
+      const healths = await Promise.all([
+        this.queueServiceHealthIndicator.isHealthy(),
+        this.triggerQueueServiceHealthIndicator.isHealthy(),
+        this.wsQueueServiceHealthIndicator.isHealthy(),
+      ]);
+
+      return healths.every((health) => !!health === true);
+    } catch (error) {
+      Logger.error(
+        'Some health indicator throw an error when checking if queues are enabled',
+        LOG_CONTEXT,
+        error
+      );
+
+      return false;
+    }
+  }
+
+  async pauseWorkers(workers: INovuWorker[]): Promise<void> {
+    for (const worker of workers) {
+      Logger.log(`Pausing worker ${worker.name}...`, LOG_CONTEXT);
+      await worker.pauseWorker();
```

Comment:
I assume that pausing might take time while finishing processing inflights calls, maybe we should do Promise.all here?

Reply:
Not sure if you are thinking this is for the graceful shutdown. This is used here: https://github.com/novuhq/novu/pull/3396/files#r1193495088

So it is meant to be used before we even accept calls (before app is ready to listen).

Reply:
Had to revert this change back as I wasn't able to find the reason tests were not working fully. We can try it in the future.